### PR TITLE
Reddit.js cleanup

### PR DESCRIPTION
**Pull request by not-an-aardvark**
Got rid of a lot of redundant code.

### CODE REVIEW

File: api/services/Reddit.js [hunk 1]
```diff
@@ -8,14 +8,15 @@ exports.data = {
   clientID: sails.config.reddit.clientID,
   clientIDSecret: sails.config.reddit.clientIDSecret,
   redirectURL: sails.config.reddit.redirectURL,
-  adminRefreshToken: sails.config.reddit.adminRefreshToken
+  adminRefreshToken: sails.config.reddit.adminRefreshToken,
+  userAgent: "Webpage:hq.porygon.co:v" + sails.config.version;
```
Comment (not-an-aardvark) [inline api/services/Reddit.js]: I changed this to something a bit more descriptive, but we can change it back if necessary

**Comment by YaManicKill** [comment]
It looks good, but I think I need to download this and look at it properly. I don't see any big issues with it just now though.

Once this and babeljs are done, I want to promise and async/await ify it. No more of those stupid callbacks.

**Comment by not-an-aardvark** [comment]
Rebased since we merged that other PR with the small improvements.

File: api/services/Reddit.js [hunk 2]
```diff
@@ -1,38 +1,26 @@
-//
 var request = require("request"),
   moment = require('moment'),
   left = 600,
-  resetTime = moment().add(600, "seconds");
+  resetTime = moment().add(600, "seconds"),
+  userAgent = "Webpage:hq.porygon.co:v" + sails.config.version;
```
Comment (not-an-aardvark) [inline api/services/Reddit.js]: I changed the user-agent string to be a bit more descriptive (it was hard-coded as `fapp/1.0` before). Now it updates with the version number. I used [these guidelines](https://github.com/reddit/reddit/wiki/API) to make this user-agent string, but it's up for modification.

Comment (YaManicKill) [inline api/services/Reddit.js]: Yeah, that sounds good. I forgot to update this sometime :-P

**Comment by not-an-aardvark** [comment]
Rebased again.

**Comment by YaManicKill** [comment]
Rebase this and I'll have a play with it tonight make sure I'm happy with it.

**Pull request merged**